### DISCUSSION TITLE
Creator failure - follow on from #33

### DISCUSSION THREAD

**Discussion by youca04** [mapsforge/mapsforge-creator#36] [Q&A] [answered]
I have started with a fresh install of Ubuntu to try and generate my own V4 maps until I can get our suppliers to upgrade their software to use the V5 maps, and I have hit the same problem as is being seen in discussion #33 in this repo. It's marked with an answer, so I thought best to open a new discussion as it didn't fix my problem.

I first saw the error after using 

```
./map-creator.sh europe/germany/berlin ram en,de,fr,es 1
```
So I tried calling osmosis directly, rather than from the script:

```
~/bin/osmosis/bin/osmosis --rb ./berlin-latest.osm.pbf --poi-writer cy.poi
```
but, this failed the same way (partial stack trace below)

```
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy


sysadmin@VM-MAPFRGE-LINUX:~/mapsforge/data/europe/germany/berlin$ java --version
openjdk 11.0.17 2022-10-18
OpenJDK Runtime Environment (build 11.0.17+8-post-Ubuntu-1ubuntu222.04)
OpenJDK 64-Bit Server VM (build 11.0.17+8-post-Ubuntu-1ubuntu222.04, mixed mode, sharing)

osmosis is osmosis-0.48.3.tgz

mapsforge-map-writer-master-20230116.114725-480-jar-with-dependencies.jar

mapsforge-poi-writer-master-20230116.114725-480-jar-with-dependencies.jar

sysadmin@VM-MAPFRGE-LINUX:~/mapsforge/data/europe/germany/berlin$ ~/bin/osmosis/bin/osmosis --rb ./berlin-latest.osm.pbf --poi-writer cy.poi
Jan 31, 2023 2:51:34 PM org.openstreetmap.osmosis.core.Osmosis run
INFO: Osmosis Version 0.48.3
Jan 31, 2023 2:51:34 PM org.openstreetmap.osmosis.core.Osmosis run
INFO: Preparing pipeline.
Jan 31, 2023 2:51:34 PM org.mapsforge.poi.writer.logging.LoggerWrapper log
INFO: POI writer version: mapsforge-poi-writer-master-SNAPSHOT
Jan 31, 2023 2:51:34 PM org.mapsforge.poi.writer.logging.LoggerWrapper log
INFO: POI format specification version: 2
Jan 31, 2023 2:51:34 PM org.mapsforge.poi.writer.logging.LoggerWrapper log
INFO: Loading categories...
Jan 31, 2023 2:51:34 PM org.mapsforge.poi.writer.XMLPoiCategoryManager <init>
SEVERE: Could not load POI category configuration from XML.
javax.xml.bind.UnmarshalException: unexpected element (uri:"http://mapsforge.org/poi-mapping", local:"category"). Expected elements are <{}category>,<{}mapping>
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:714)
        at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:232)
        at com.sun.xml.bind.v2.runtime.unmarshaller.Loader.reportError(Loader.java:227)
// Cut
Jan 31, 2023 2:51:34 PM org.openstreetmap.osmosis.core.Osmosis main
SEVERE: Execution aborted.
java.lang.NullPointerException
        at org.mapsforge.poi.writer.XMLPoiCategoryManager.<init>(XMLPoiCategoryManager.java:70)


```

**Comment by devemux86** [comment]
Try with [Osmosis 0.47](https://bretth.dev.openstreetmap.org/osmosis-build) as mentioned in the answer of that discussion.

Testing with Osmosis 0.47 and OpenJDK Temurin 11 on Debian, the map-writer and poi-writer seem to work properly.
(using the latest snapshot jar-with-dependencies: [map-writer](https://oss.sonatype.org/content/repositories/snapshots/org/mapsforge/mapsforge-map-writer/master-SNAPSHOT/) and [poi-writer](https://oss.sonatype.org/content/repositories/snapshots/org/mapsforge/mapsforge-poi-writer/master-SNAPSHOT/))

Or you can also try the suggestion mentioned [there](https://github.com/mapsforge/mapsforge-creator/discussions/33#discussioncomment-846863).

**Comment by devemux86** [comment]
Testing with Osmosis 0.48.3 it also works.

So it should be something in local setup, probably the Java edition.
Try with [OpenJDK Temurin 11](https://adoptium.net/temurin/releases/?version=11).

**Reply by youca04** [reply]
Hi,

Thanks for getting back to me so soon. I tried back rev to 0.47, which didn't work, and then also tried the Temurin OpenJD (output below).

The downloaded XSD and the embedded XML look fine, so I guess it's environmental to Ubuntu, which is also common to the other issue.
I tried increasing logging with Osmosis, but didn't show me much more so need to mess around with advanced logging or strace!

```
sysadmin@VM-MAPFRGE-LINUX:~/mapsforge/data/europe/germany/berlin$ java --version
openjdk 11.0.18 2023-01-17
OpenJDK Runtime Environment Temurin-11.0.18+10 (build 11.0.18+10)
OpenJDK 64-Bit Server VM Temurin-11.0.18+10 (build 11.0.18+10, mixed mode)
sysadmin@VM-MAPFRGE-LINUX:~/mapsforge/data/europe/germany/berlin$ ~/bin/osmosis/bin/osmosis --rb ./berlin-latest.osm.pbf --poi-writer cy.poi
Jan 31, 2023 4:03:33 PM org.openstreetmap.osmosis.core.Osmosis run
INFO: Osmosis Version 0.47
Jan 31, 2023 4:03:33 PM org.openstreetmap.osmosis.core.Osmosis run
INFO: Preparing pipeline.
Jan 31, 2023 4:03:33 PM org.mapsforge.poi.writer.logging.LoggerWrapper log
INFO: POI writer version: mapsforge-poi-writer-master-SNAPSHOT
Jan 31, 2023 4:03:33 PM org.mapsforge.poi.writer.logging.LoggerWrapper log
INFO: POI format specification version: 2
Jan 31, 2023 4:03:33 PM org.mapsforge.poi.writer.logging.LoggerWrapper log
INFO: Loading categories...
Jan 31, 2023 4:03:33 PM org.mapsforge.poi.writer.XMLPoiCategoryManager <init>
SEVERE: Could not load POI category configuration from XML.
javax.xml.bind.UnmarshalException: unexpected element (uri:"http://mapsforge.org/poi-mapping", local:"category"). Expected elements are <{}category>,<{}mapping>
        at com.sun.xml.bind.v2.runtime.unmarshaller.UnmarshallingContext.handleEvent(UnmarshallingContext.java:714)
```

**Comment by devemux86** [comment, marked as answer]
In the other discussion they used external poi-mapping and map-mapping xml files.
They edited them and removed a definition and it worked, see https://github.com/mapsforge/mapsforge-creator/discussions/33#discussioncomment-846863.

**Reply by youca04** [reply]
I am going to try that - need to work out how to pass these through the script and into the JVM, etc. Thanks again.

**Reply by youca04** [reply]
Hi @devemux86 . Thanks for your assistance.

Removing the namespace and XSD from the poi-mapping.xml got me past the POI error.

I used the override to point to a modified version of the file.
`export POI_TAG_CONF_FILE=/home/sysadmin/Downloads/poi-mapping.xml`

When I get a chance, I will work out what is causing Xerces to choke on the XSD with a small test application. I've seen similar behaviour on other projects, but I cannot remember what caused it offhand.

I also managed to generate the v4 maps successfuly.
